artiq/gateware/rtio/phy/dds.py

```diff
@@ -140,12 +141,12 @@ def __init__(self, current_data, data_length, flags, ch_sel):
             If(ch_sel,
                 If(flags & SPI_END,
                     # lower 16 bits (16-32 from 48-bit transfer)
-                    NextValue(self.ftw[0:16], current_data[16:32]),
+                    NextValue(self.ftw[:16], current_data[16:]),
                     NextState("START")
-                ).Else(
-                    # upper 16 bits (32-48 from 48-bit transfer)
-                    NextValue(self.ftw[16:32], current_data[0:16]),
-            ))
+                ).Elif(current_data != 0,
```

Comment:
Since ``current_data`` is connected to ``rtlink.o.data`` through comb instead of getting latched, the ftw value would get overwritten with zeroes. Checking the value like that is a bit of an ugly workaround and while it works, I'd rather do it properly, with a register.